e2etest/newe2e_task_validation.go

```diff
@@ -191,3 +194,110 @@ func checkMultipleErrors(errorMsg []string, line string) bool {
 
 	return false
 }
+
+func ValidateListTextOutput(a Asserter, stdout AzCopyStdout, expectedObjects map[AzCopyOutputKey]cmd.AzCopyListObject, expectedSummary *cmd.AzCopyListSummary) {
+	if dryrunner, ok := a.(DryrunAsserter); ok && dryrunner.Dryrun() {
+		return
+	}
+
+	expectedObj := expectedObjectsToString(expectedObjects)
+	rawObj, rawSum := cleanupRaw(stdout.RawStdout(), expectedSummary != nil)
+
+	// sort to avoid out of order arrays
+	sort.Strings(rawObj)
+	sort.Strings(expectedObj)
+
+	if !reflect.DeepEqual(rawObj, expectedObj) {
+		a.Error(fmt.Sprintf("Does not match - raw:%s. expectedObj:%s.", rawObj, expectedObj))
+	}
+
+	// if expectedObj summary is provided, the last two elements of raw are part of the summary
+	if expectedSummary != nil {
+		validateSummary(a, rawSum, expectedSummary)
+	}
+}
+
+func cleanupRaw(raw []string, hasSummary bool) ([]string, []string) {
```

Comment:
`cleanupRaw` is not a super great function name either. I wouldn't describe what's happening here as "cleaning up", and this function is an extremely verbose (and slow!) way to do `output, summary := raw[:len(raw)-summaryLength], raw[len(raw)-summaryLength]`.